keydata_test.go

```diff
@@ -448,70 +526,111 @@ func (s *keyDataSuite) checkKeyDataJSONAuthModeNone(c *C, keyData *KeyData, crea
 	s.checkKeyDataJSONFromReaderAuthModeNone(c, w.Reader(), creationParams, nmodels)
 }
 
-func (s *keyDataSuite) checkKeyDataJSONAuthModePassphrase(c *C, keyData *KeyData, creationParams *KeyParams, nmodels int, passphrase string, kdfOpts *KDFOptions) {
+func (s *keyDataSuite) checkKeyDataJSONAuthModePassphrase(c *C, keyData *KeyData, creationParams *KeyWithPassphraseParams, nmodels int, passphrase string, kdfOpts *KDFOptions) {
 	w := makeMockKeyDataWriter()
 	c.Check(keyData.WriteAtomic(w), IsNil)
 
 	s.checkKeyDataJSONFromReaderAuthModePassphrase(c, w.Reader(), creationParams, nmodels, passphrase, kdfOpts)
 }
 
 type testKeyPayloadData struct {
-	key    DiskUnlockKey
-	auxKey AuxiliaryKey
+	primary PrimaryKey
+	unique  []byte
+}
+
+func marshalASN1(c *C, primary PrimaryKey, unique []byte) []byte {
+	builder := cryptobyte.NewBuilder(nil)
+
+	builder.AddASN1(cryptobyte_asn1.SEQUENCE, func(b *cryptobyte.Builder) { // ProtectedKeys ::= SEQUENCE {
+		b.AddASN1OctetString(primary) // primary OCTETSTRING
+		b.AddASN1OctetString(unique)  // unique OCTETSTRING
+	})
+
+	b, err := builder.Bytes()
+	c.Assert(err, IsNil)
+	return b
 }
 
 func (s *keyDataSuite) testKeyPayload(c *C, data *testKeyPayloadData) {
-	payload := MarshalKeys(data.key, data.auxKey)
+	payload := marshalASN1(c, data.primary, data.unique)
 
-	key, auxKey, err := payload.Unmarshal()
+	pk, err := UnmarshalProtectedKeys(payload)
 	c.Check(err, IsNil)
-	c.Check(key, DeepEquals, data.key)
-	c.Check(auxKey, DeepEquals, data.auxKey)
+
+	dj := new(bytes.Buffer)
+	json.NewEncoder(dj).Encode(data)
+
+	pkj := new(bytes.Buffer)
+	json.NewEncoder(pkj).Encode(pk)
+
+	c.Check(dj.Bytes(), DeepEquals, pkj.Bytes())
```

Comment:
Rather than encoding `protectedKeys` to JSON in order to byte compare them, I would:
- Allow the members of `protectedKeys` to be exported (ie, rename them to `Primary` and `Unique`.
- Add a type alias to export_test.go (`type ProtectedKeys = protectedKeys`)
- Do `c.Check(pk, DeepEquals, &ProtectedKeys{Primary: data.primary, Unique: data.unique})`

Reply:
Thanks for the recommendation, I fixed it as suggested :) 

I was not sure what was your intention with making them private so I thought not to change them.

Reply:
Thanks, that looks good.

The members are still private by default because the structure isn't exported - all renaming them does here is make them exportable so that they are accessible by adding an exported type alias just for testing. I generally keep members of unexported types unexportable unless it's useful to have direct access to them in unit tests.